src/Cli.Tests/StringLogger.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Cli.Tests;
+
+/// <summary>
+/// Creates a logger that can be used in test methods to verify logging behavior
+/// by capturing the messages and making them available for verification.
+/// </summary>
+internal class StringLogger : ILogger
```

Comment:
nit: I believe within a namespace, classes are declared as internal by default so can omit the internal keyword here.

Reply:
[You are correct}(https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers#class-record-and-struct-accessibility), I'll address that.


```suggestion
class StringLogger : ILogger
```